pulsar-broker/src/main/java/org/apache/pulsar/broker/stats/prometheus/PrometheusMetricsServlet.java

```diff
@@ -70,18 +71,27 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
         AsyncContext context = request.startAsync();
         context.setTimeout(metricsServletTimeoutMs);
         executor.execute(safeRun(() -> {
+            long start = System.currentTimeMillis();
             HttpServletResponse res = (HttpServletResponse) context.getResponse();
             try {
                 res.setStatus(HttpStatus.OK_200);
                 res.setContentType("text/plain");
                 PrometheusMetricsGenerator.generate(pulsar, shouldExportTopicMetrics, shouldExportConsumerMetrics,
                         shouldExportProducerMetrics, splitTopicAndPartitionLabel, res.getOutputStream(),
                         metricsProviders);
-                context.complete();
-
             } catch (Exception e) {
-                log.error("Failed to generate prometheus stats", e);
+                long end = System.currentTimeMillis();
+                long time = end - start;
```

Comment:
I wonder if we should use Guava's Stopwatch for such use cases? Javadoc: https://guava.dev/releases/snapshot-jre/api/docs/com/google/common/base/Stopwatch.html

Reply:
I don't like very much Guava Stopwatch (in the past the broke compatibility), and in general I believe that we should keep off Guava as much as possible (still for the fact that they break the APIs quite often)